borg/helpers.py

```diff
@@ -151,7 +152,13 @@ def load(cls, repository, key=None, force_tam_not_required=False):
     def write(self):
         if self.key.tam_required:
             self.config[b'tam_required'] = True
-        self.timestamp = datetime.utcnow().isoformat()
+        # self.timestamp needs to be strictly monotonically increasing. Clocks often are not set correctly
+        if self.timestamp is None:
+            self.timestamp = datetime.utcnow().isoformat()
+        else:
+            prev_ts = datetime.strptime(self.timestamp, "%Y-%m-%dT%H:%M:%S.%f")
+            incremented = (prev_ts + timedelta(microseconds=1)).isoformat()
+            self.timestamp = max(incremented, datetime.utcnow().isoformat())
```

Comment:
Yes, that is indeed the case. But the current behavior is worse.
The only secure way i see is to introduce an epoch that is incremented instead of the timestamp or maybe prepended if the time is too much in the future. But of course the device with the bad clock can also be the device doing the current operation so the epoch might get quite high.

I actually thought the rollback protection would use just an number that is incremented but when looking at some other code i was reminded that it is a date. Technically there is no need to have a correct date. I didn‘t find any place where it is actually output or used as a timestamp, so i don‘t think we need to invest too much energy into keeping it the same as the current time.